tutor/src/flux/task-plan-stats.coffee

```diff
@@ -8,15 +8,18 @@ TaskPlanStatsConfig = {
     getPeriods: (id) ->
       plan = @_get(id)
       periods = _.chain(plan.stats)
-        .map((stat) ->
+        # filter out deleted periods
+        .filter (stat) -> stat.period_id?
+        .map (stat) ->
           id: stat.period_id
           name: stat.name
           is_trouble: stat.is_trouble
-
-        )
-        .tap(PeriodHelper.sort)
         .value()
 
+      # tapping into the sort wasn't actually returning sorted
+      # periods for some reason
+      PeriodHelper.sort(periods)
+
```

Comment:
Pretty sure that's because `.tap` doesn't actually use the return value of the function. It's intended to allow a function to manipulate the values, but not change them to a new thing.  If you wanted to set a new order for periods, I think you could just use `
`.sort(PeriodHelper.sort)`  This works though - is equally good to me